domain/src/test/java/org/oppia/android/domain/spotlight/SpotlightStateControllerTest.kt

```diff
@@ -0,0 +1,388 @@
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(retrieveSpotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_invalidFeature_returnsFailure() {
+    val invalidFeature = Spotlight.FeatureCase.FEATURE_NOT_SET
+    val spotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId, invalidFeature)
+    dataProviderTestMonitor.waitForNextFailureResult(spotlightStateProvider)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_validFeature_marked_differentProfile_returnsSpotlightSeen() {
+    dataProviderTestMonitor.waitForNextSuccessfulResult(
+      spotlightStateController.markSpotlightViewed(
+        profileId1,
+        FIRST_CHAPTER
+      )
+    )
+
+    val spotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId1, FIRST_CHAPTER)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(spotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_SEEN)
+  }
+
+  @Test
+  fun testRetrieveSpotlightViewState_validFeature_notMarked_differentProfile_returnsSpotlightNotSeen() { // ktlint-disable max-line-length
+    val spotlightStateProvider =
+      spotlightStateController.retrieveSpotlightViewState(profileId1, FIRST_CHAPTER)
+    val result = dataProviderTestMonitor.waitForNextSuccessfulResult(spotlightStateProvider)
+    assertEquals(result, SpotlightViewState.SPOTLIGHT_NOT_SEEN)
+  }
+
+  private fun markSpotlightSeen(spotlightFeature: Spotlight.FeatureCase) {
+    dataProviderTestMonitor.waitForNextSuccessfulResult(
+      spotlightStateController.markSpotlightViewed(
+        profileId,
+        spotlightFeature
+      )
+    )
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>()
+      .inject(this)
+  }
+
+  @Module
+  class TestModule {
+    @Provides
+    @Singleton
+    fun provideContext(application: Application): Context {
+      return application
+    }
+
+    @EnableConsoleLog
+    @Provides
+    fun provideEnableConsoleLog(): Boolean = true
+
+    @EnableFileLog
+    @Provides
+    fun provideEnableFileLog(): Boolean = false
+
+    @GlobalLogLevel
+    @Provides
+    fun provideGlobalLogLevel(): LogLevel = LogLevel.VERBOSE
+
+    @CacheAssetsLocally
+    @Provides
+    fun provideCacheAssetsLocally(): Boolean = false
+
+    @Provides
+    @TopicListToCache
+    fun provideTopicListToCache(): List<String> = listOf()
+
+    @Provides
+    @LoadLessonProtosFromAssets
+    fun provideLoadLessonProtosFromAssets(testEnvironmentConfig: TestEnvironmentConfig): Boolean =
+      testEnvironmentConfig.isUsingBazel()
+
+    @Provides
+    @LearnerStudyAnalytics
+    fun provideLearnerStudyAnalytics(): PlatformParameterValue<Boolean> {
+      // Enable the study by default in tests.
+      return PlatformParameterValue.createDefaultParameter(defaultValue = true)
```

Comment:
Is this necessary? The parameter doesn't seem to actually affect the controller from what I can see. Could we just depend on the test parameter module to simplify things a bit?

Reply:
done